src/plugins/installer.py

```diff
@@ -0,0 +1,96 @@
+import os
+import pathlib
+
+from helperFunctions.install import run_cmd_with_logging
+
+
+class AbstractPluginInstaller():
+    # Even if some functoins don't need self we want to have them nicely
+    # grouped in this class
+    # pylint:disable=no-self-use
+    def __init__(self, base_path, distribution, skip_docker=False):
+        super().__init__()
+        self.base_path = base_path
+        self.distribution = distribution
+        self.build_path = base_path / pathlib.Path("build")
+        self.skip_docker = skip_docker
+
+    def install(self):
+        cwd = os.getcwd()
+        os.chdir(self.base_path)
+        self.system_packages()
+        self.pip_packages()
+        self.other_packages()
+
+        self.files()
+
+        self.build_path.mkdir(exist_ok=True)
+        os.chdir(self.build_path)
+        self.build()
+        os.chdir(self.base_path)
+
+        if not self.skip_docker:
+            self.docker()
+
+        self.do_last()
+
+        os.chdir(cwd)
+
+    def docker(self):
+        """
+        Build/Pull docker images
+        """
+
+    def system_packages(self):
+        """
+        Install packages with apt/dnf
+        """
+        build_pkg_path = "./apt-pkgs-build.txt" if self.distribution != "fedora" else "./dnf-pkgs-build.txt"
+        runtime_pkg_path = "./apt-pkgs-runtime.txt" if self.distribution != "fedora" else "./dnf-pkgs-runtime.txt"
+
+        build_list = []
+        if os.access(build_pkg_path, os.R_OK):
+            with open(build_pkg_path) as build_f:
+                build_list = build_f.read().splitlines()
+
+        runtime_list = []
+        if os.access(runtime_pkg_path, os.R_OK):
+            with open(runtime_pkg_path) as runtime_f:
+                runtime_list = runtime_f.read().splitlines()
+
+        pgk_mgr_cmd = "apt install -y" if self.distribution != "fedora" else "dnf install -y"
+        pkgs_to_install = build_list + runtime_list
+
+        if len(pkgs_to_install) == 0:
+            return
+
+        run_cmd_with_logging(f"sudo {pgk_mgr_cmd} " + " ".join(pkgs_to_install))
+
+    def pip_packages(self):
+        """
+        Install packages with pip
+        """
+        if os.access("./requirements.txt", os.R_OK):
```

Comment:
IMHO pathlib improves readability a bit: 
```suggestion
        if Path("./requirements.txt").exists():
```